### PR TITLE
chore: improve overall flow

### PR DESCRIPTION
**Pull request by linuxbandit**
README updated with moving parts section
Better flow for orchestration, no waste of time due to dos/unix encoding, no sudo on machine
Install npx on machine
Add make targets to rebuild only specific backends

### CODE REVIEW

File: vagrant-post-script/orchestrate_docker.sh [hunk 1]
```diff
@@ -1,11 +1,16 @@
 #!/bin/bash
 
 cd /vagrant/
-#git reset --hard HEAD
-#./helper.sh --bumpmodules
+
+#install nvm
+curl -sS -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash
```
Comment (serge1peshcoff) [inline vagrant-post-script/orchestrate_docker.sh]: I'd also install node there

Comment (linuxbandit) [reply vagrant-post-script/orchestrate_docker.sh]: I think it is installed by nvm? https://github.com/nvm-sh/nvm/blob/v0.35.3/install.sh#L67

Comment (serge1peshcoff) [reply vagrant-post-script/orchestrate_docker.sh]: Can you try running node -v after installing to check if it works?

Comment (linuxbandit) [reply vagrant-post-script/orchestrate_docker.sh]: no, I have no free space for the VM. @LeonVreling can you try?
`git pull` (inside oms-docker)
`git checkout chore/various-improvements`
`vagrant up --provision`

Comment (linuxbandit) [reply vagrant-post-script/orchestrate_docker.sh]: or whoever is free earlier between @maaikeheijdenrijk @mibaru-aegee

Comment (linuxbandit) [reply vagrant-post-script/orchestrate_docker.sh]: anyway I went lower in the code and yes, it does install node automatically, I think we don't need to check it (what good would an nvm do without node? imho it's obvious it's installed)

Comment (serge1peshcoff) [reply vagrant-post-script/orchestrate_docker.sh]: nice then

File: Makefile [hunk 2]
```diff
@@ -24,6 +24,21 @@ refresh:  build
...
+listen_frontend:
+	cd oms-frontend && npm run build -- --watch
+
+rebuild_frontend:
+	./helper.sh --docker -- up -d --build --force-recreate oms-frontend
+
+rebuild_core:
+	./helper.sh --docker -- up -d --build --force-recreate oms-core-elixir
+
+rebuild_events:
```
Comment (serge1peshcoff) [inline Makefile:36]: also mailer

File: vagrant-post-script/orchestrate_docker.sh [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
 
 cd /vagrant/
-#git reset --hard HEAD
-#./helper.sh --bumpmodules
+
 if [[ -f .env ]]; then
+  dos2unix .env
   make start
 else
+  git reset --hard HEAD
```
Comment (serge1peshcoff) [inline vagrant-post-script/orchestrate_docker.sh:9]: what's this line for?

Comment (linuxbandit) [reply vagrant-post-script/orchestrate_docker.sh:9]: encoding change between dos and unix

Comment (serge1peshcoff) [reply vagrant-post-script/orchestrate_docker.sh:9]: I mean `git reset` line

Comment (linuxbandit) [reply vagrant-post-script/orchestrate_docker.sh:9]: by resetting, the wrong encoding disappears

Comment (serge1peshcoff) [reply vagrant-post-script/orchestrate_docker.sh:9]: weird, but ok

**Review by serge1peshcoff — APPROVED** [review]

File: vagrant-post-script/orchestrate_docker.sh (hunk 1 above)
Comment (linuxbandit) [reply vagrant-post-script/orchestrate_docker.sh]: v12.14.0 (leon jansen confirmed that)

**Pull request merged**